langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/FromHexString.java

```diff
@@ -29,6 +35,11 @@
 
     public static Object fromHexString(BString s) {
         // TODO: 6/20/19 Implement this
-        return null;
+        try {
+            return Double.parseDouble(s.getValue());
```

Comment:
Is [java's hex float format](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#valueOf(java.lang.String)) same as Ballerina spec's format?

If yes please all test cases that covers all [production](https://en.wikipedia.org/wiki/Production_(computer_science)) rules.

Reply:
Yes, the Java's [HexFloatingPointLiteral](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#valueOf(java.lang.String)) has a similar format to the [HexFloatingPointLiteral](https://ballerina.io/spec/lang/2021R1/#HexIndicator). And I've added some test cases to cover the cases mentioned in the [spec](https://ballerina.io/spec/lang/2021R1/#section_5.2.4)